This is a test fot the 101-lazy_matrix_mul module
===================================================
This is a doctest file for the lazy_matrix_mul function
----------------------------------------------------
>>> lazy_matrix_mul = __import__('101-lazy_matrix_mul').lazy_matrix_mul

>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]])
array([[ 7, 10],
       [15, 22]])

>>> lazy_matrix_mul([[1, 2]], [[3, 4], [5, 6]])
array([[13, 16]])

>>> lazy_matrix_mul("Error", [[5, 6], [7, 8]])
Traceback (most recent call last):
    ...
numpy.core._exceptions._UFuncNoLoopError: ufunc 'matmul' did not contain a loop with signature matching types (dtype('<U5'), dtype('int64')) -> None

>>> lazy_matrix_mul([1, 2, 3, 4], "Fibbing")
Traceback (most recent call last):
    ...
numpy.core._exceptions._UFuncNoLoopError: ufunc 'matmul' did not contain a loop with signature matching types (dtype('int64'), dtype('<U7')) -> None

>>> lazy_matrix_mul([1, 2], [[3, 4], [5, 6]])
array([13, 16])

>>> lazy_matrix_mul([[1, 2], [3, 4]], [5, 6])
array([17, 39])

>>> lazy_matrix_mul([[1, 2, 3], [3, 4, 5]], [[1, 2], [3, 4]])
Traceback (most recent call last):
...
ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 2 is different from 3)

>>> lazy_matrix_mul([[5, 6], [7, 8]], [[5, 6, 1], [7, 8]])
Traceback (most recent call last):
...
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
